From cde1ed44f86c3521b7f058c7a7476e423a1a1511 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Thu, 23 Mar 2017 18:30:39 -0400
Subject: [PATCH 29/86] prp: fix LSDU size calculation

Currently, driver includes the Tag size twice in the calculation
of LSDU size. This patch fixes the same.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 net/hsr-prp/hsr_prp_forward.c | 2 +-
 net/hsr-prp/hsr_prp_main.h    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/hsr-prp/hsr_prp_forward.c b/net/hsr-prp/hsr_prp_forward.c
index 5f3479a..674a49c 100644
--- a/net/hsr-prp/hsr_prp_forward.c
+++ b/net/hsr-prp/hsr_prp_forward.c
@@ -205,7 +205,7 @@ static void prp_fill_rct(struct sk_buff *skb,
 		skb_put_padto(skb, ETH_ZLEN);
 
 	trailor = (struct prp_rct *)skb_put(skb, HSR_PRP_HLEN);
-	lsdu_size = skb->len - 14 + HSR_PRP_HLEN;
+	lsdu_size = skb->len - 14;
 	if (frame->is_vlan)
 		lsdu_size -= 4;
 	prp_set_lan_id(trailor, port);
diff --git a/net/hsr-prp/hsr_prp_main.h b/net/hsr-prp/hsr_prp_main.h
index 54f8aca..d34174d 100644
--- a/net/hsr-prp/hsr_prp_main.h
+++ b/net/hsr-prp/hsr_prp_main.h
@@ -295,7 +295,7 @@ static inline bool prp_check_lsdu_size(struct sk_buff *skb,
 		expected_lsdu_size = HSR_PRP_V1_SUP_LSDUSIZE;
 	} else {
 		ethhdr = (struct ethhdr *)skb_mac_header(skb);
-		expected_lsdu_size = skb->len - 14 + HSR_PRP_HLEN;
+		expected_lsdu_size = skb->len - 14;
 		if (ethhdr->h_proto == htons(ETH_P_8021Q))
 			expected_lsdu_size -= 4;
 	}
-- 
1.9.1

